common/export-common-cartridge-util/src/test/org/sakaiproject/exporter/util/ExportUtilTest.java

```diff
@@ -0,0 +1,174 @@
+		StringBuilder result = exportUtil.addPDFViewer(source, links);
+		assertEquals("<a href='/group/" + siteId + "/file.txt'>File</a>", result.toString());
+	}
+
+	@Test
+	public void testUpperCase() {
+		links.put("file.PDF", false);
+		StringBuilder source = new StringBuilder("<a href='/group/" + siteId + "/file.PDF'>File</a>");
+		StringBuilder result = exportUtil.addPDFViewer(source, links);
+		assertThat(result.toString(), Matchers.containsString("/group/" + siteId + "/file.PDF?canvas_download"));
+	}
+
+	@Test
+	public void testSpaceInFile() {
+		links.put("file%20name.pdf", false);
+		StringBuilder source = new StringBuilder("<a href='/group/" + siteId + "/file%20name.pdf'>File</a>");
+		StringBuilder result = exportUtil.addPDFViewer(source, links);
+		assertThat(result.toString(), Matchers.containsString("/group/" + siteId + "/file%20name.pdf?canvas_download"));
+	}
+
+	@Test
+	public void testAdditionalAttributes() {
+		links.put("file.pdf", false);
+		StringBuilder source = new StringBuilder("<a href='/group/" + siteId + "/file.pdf' title='Hello &amp; Bye'>File</a>");
+		StringBuilder result = exportUtil.addPDFViewer(source, links);
+		assertThat(result.toString(), Matchers.containsString("/group/" + siteId + "/file.pdf?canvas_download"));
+	}
+
+	@Test
+	public void testMoreCharacters() {
+		links.put("file.pdf", false);
+		StringBuilder source = new StringBuilder("<a href='/group/" + siteId + "/file.pdf' title='+++'>File</a>");
+		StringBuilder result = exportUtil.addPDFViewer(source, links);
+		assertThat(result.toString(), Matchers.containsString("/group/" + siteId + "/file.pdf?canvas_download"));
+	}
+
+	//@Test
+	// Not sure if we should bother here
+	public void testExistingClass() {
+		links.put("file.pdf", false);
+		StringBuilder source = new StringBuilder("<a class='myclass' href='/group/" + siteId + "/file.pdf' >File</a>");
+		StringBuilder result = exportUtil.addPDFViewer(source, links);
+		assertEquals("<a class=\"myclass instructure_file_link instructure_scribd_file\"" +
+				" href='/group/" + siteId + "/file.pdf?canvas_download=1&amp;canvas_qs_wrap=1' >File</a>", result.toString());
+	}
+
+	@Test
+	public void testFilePathAnotherSite() {
+		String path = "/group/wrongSiteId/file.pdf";
+		String result = exportUtil.getFilePath(path, false);
+		assertEquals(result, path); 
+	}
+
+	@Test
+	public void testFilePathFlat() {
+		String path = "/group/" + siteId + "/file.pdf";
+		String result = exportUtil.getFilePath(path, true);
+		assertEquals("", result);
+	}
+
+	@Test
+	public void testFilePathEscaped() {
+		String path = "/group/" + siteId + "/dot&dash/ürm/file.pdf";
+		String result = exportUtil.getFilePath(path, true);
+		assertEquals("dot%26dash/%C3%BCrm/", result);
+	}
+
+	//@Test
+	public void testRemoveReadingLists() {
+		ContentResource cr = mock(ContentResource.class);
+		files.put("IsCitation", cr);
+
+		when(cr.getContentType()).thenReturn("org.sakaiproject.citation.impl.CitationList");
```

Comment:
I think this should be `cr.getResourceType()`

Reply:
I've made this change and enabled the test.